crates/torii/src/engine.rs

```diff
@@ -50,13 +51,15 @@ impl<'a, S: State + Executable, T: JsonRpcTransport + Sync + Send> Engine<'a, S,
         storage: &'a S,
         provider: &'a JsonRpcClient<T>,
         processors: Processors<S, T>,
+        start_block: Option<u64>,
         config: EngineConfig,
     ) -> Self {
-        Self { storage, provider, processors, config }
+        Self { storage, provider, processors, start_block, config }
     }
 
     pub async fn start(&self) -> Result<(), Box<dyn Error>> {
-        let mut current_block_number = self.storage.head().await?;
+        let mut current_block_number =
+            if let Some(head) = self.start_block { head } else { self.storage.head().await? };
```

Comment:
hmm will this restart from the provided start block of the sync has already started? perhaps we want to make sure the stored head is 0?